eth/contracts/facets/DFPlayerProfileFacet.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+// Library imports
+import {LibDiamond} from "../vendor/libraries/LibDiamond.sol";
+
+// Storage imports
+import {WithStorage, PlayerProfileStorage, LibStorage} from "../libraries/LibStorage.sol";
+
+contract DFPlayerProfileFacet is WithStorage {
+  event PlayerDisplayNameUpdated(address player, string displayName);
+
+  // Struct to hold address and associated player profile.
+  struct PlayerProfile {
+    address playerAddress;
+    string displayName;
+  }
+
+  // Modifier to control access
+  modifier onlyAdminOrSelf(address player) {
+    require(msg.sender == LibDiamond.contractOwner() || msg.sender == player, "Not authorized");
+    _;
+  }
+
+  // Function to set the displayName, accessible only by admin or player themselves
+  function setDisplayName(address player, string memory newDisplayName) public onlyAdminOrSelf(player) {
+    PlayerProfileStorage storage pps = LibStorage.playerProfileStorage();
```

Comment:
function pps() internal pure returns (PlayerProfileStorage storage) {
        return LibStorage.playerProfileStorage();
}

You can directly use pps()